### DISCUSSION TITLE
KIA EV6, Charge level

### DISCUSSION THREAD

**Discussion by chersoph** [Hyundai-Kia-Connect/kia_uvo#651] [Q&A]
In the Kia Connect app, there is information about the Battery level (Status- Charging & Climate - Battery Level). It is not the same value as the "sensor.ev6_car_battery_level", that i suspect is the 12V battery. For instance at this moment 98% for sensor.ev6_car_battery_level and the app battery level is 83%.
Can the app battery level be added to the list of attributes/sensors in the integration?

**Comment by cdnninja** [comment]
Sounds like a mapping issue. Do you have your attributes from the data sensor as something you can post? Also what region and brand?

**Comment by chersoph** [comment]
Region, Sweden
Brand, KIA
Car modell, EV6

Data sensor.

vehicle_data: 
vehicleLocation:
  coord:
    lat: 57.xxxxxx
    lon: 12.xxxxxx
    alt: 0
    type: 0
  head: 0
  speed:
    value: 0
    unit: 0
  accuracy:
    hdop: 0
    pdop: 0
  time: '20230606100650'
vehicleStatus:
  airCtrlOn: false
  engine: false
  doorLock: true
  doorOpen:
    frontLeft: 0
    frontRight: 0
    backLeft: 0
    backRight: 0
  trunkOpen: false
  airTemp:
    value: 02H
    unit: 0
    hvacTempType: 1
  defrost: false
  acc: false
  evStatus:
    batteryCharge: false
    batteryStatus: 90
    batteryPlugin: 0
    remainTime2:
      etc1:
        value: 21
        unit: 1
      etc2:
        value: 280
        unit: 1
      etc3:
        value: 65
        unit: 1
      atc:
        value: 280
        unit: 1
    drvDistance:
      - rangeByFuel:
          evModeRange:
            value: 430
            unit: 1
          totalAvailableRange:
            value: 430
            unit: 1
        type: 2
    reservChargeInfos:
      reservChargeInfo:
        reservChargeInfoDetail:
          reservInfo:
            day:
              - 9
            time:
              time: '1200'
              timeSection: 0
          reservChargeSet: false
          reservFatcSet:
            defrost: false
            airTemp:
              value: 00H
              unit: 0
              hvacTempType: 1
            airCtrl: 0
            heating1: 0
      offpeakPowerInfo:
        offPeakPowerTime1:
          starttime:
            time: '1200'
            timeSection: 0
          endtime:
            time: '1200'
            timeSection: 0
        offPeakPowerFlag: 0
      reserveChargeInfo2:
        reservChargeInfoDetail:
          reservInfo:
            day:
              - 9
            time:
              time: '1200'
              timeSection: 0
          reservChargeSet: false
          reservFatcSet:
            defrost: false
            airTemp:
              value: 00H
              unit: 0
              hvacTempType: 1
            airCtrl: 0
            heating1: 0
      reservFlag: 0
      ect:
        start:
          day: 0
          time:
            time: '0000'
            timeSection: 0
        end:
          day: 0
          time:
            time: '0000'
            timeSection: 0
      targetSOClist:
        - targetSOClevel: 100
          dte:
            rangeByFuel:
              evModeRange:
                value: 483
                unit: 1
              totalAvailableRange:
                value: 483
                unit: 1
            type: 2
          plugType: 0
        - targetSOClevel: 90
          dte:
            rangeByFuel:
              evModeRange:
                value: 430
                unit: 1
              totalAvailableRange:
                value: 430
                unit: 1
            type: 2
          plugType: 1
    chargePortDoorOpenStatus: 1
    batteryPreconditioning: false
  ign3: false
  hoodOpen: false
  transCond: false
  steerWheelHeat: 0
  sideBackWindowHeat: 0
  tirePressureLamp:
    tirePressureLampAll: 0
    tirePressureLampFL: 0
    tirePressureLampFR: 0
    tirePressureLampRL: 0
    tirePressureLampRR: 0
  battery:
    batSoc: 98
    batState: 0
    sjbDeliveryMode: 0
    batSignalReferenceValue:
      batWarning: 0
    powerAutoCutMode: 2
  lampWireStatus:
    stopLamp:
      stopLampStatus: false
      leftLamp: false
      rightLamp: false
    headLamp:
      headLampStatus: false
      leftLowLamp: false
      rightLowLamp: false
      leftHighLamp: false
      rightHighLamp: false
      leftBifuncLamp: false
      rightBifuncLamp: false
    turnSignalLamp:
      turnSignalLampStatus: false
      leftFrontLamp: false
      rightFrontLamp: false
      leftRearLamp: false
      rightRearLamp: false
  windowOpen:
    frontLeft: 0
    frontRight: 0
    backLeft: 0
    backRight: 0
  smartKeyBatteryWarning: false
  washerFluidStatus: false
  breakOilStatus: false
  sleepModeCheck: false
  time: '20230606100649'
  remoteWaitingTimeAlert:
    remoteControlAvailable: 1
    remoteControlWaitingTime: 168
    elapsedTime: '16:47:48'
  systemCutOffAlert: 0
  tailLampStatus: 0
  hazardStatus: 0
odometer:
  value: 540.5
  unit: 1

vehicle_name: EV6
friendly_name: EV6 Data
